cmds/console/uart_linux.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2012-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// This file contains support functions for UART io for Linux.
+package main
+
+import (
+	"os"
+)
+
+const (
+	data   = 0x3f8
+	status = 0x3fd
+	inRdy  = 1
+	outRdy = 0x40
+)
+
+type uart struct{}
```

Comment:
This doesn't quite make sense to me.

Either have a package with a global file, but no struct,

or the struct should contain the open file and operate only on its open file.

```go

type uart struct {
  f *os.File
}

func OpenUART() (io.ReadWriter, error) {
  port, err := os.OpenFile("/dev/port", os.O_RDWR, 0)
  if err != nil {
    return err
  }
  return uart{port}
}
```